ProjFS.Mac/PrjFSKext/FileSystemMountPoints.cpp

```diff
@@ -0,0 +1,134 @@
+#include "FileSystemMountPoints.hpp"
+#include "kernel-header-wrappers/mount.h"
+#include "Locks.hpp"
+#include "KextLog.hpp"
+#include <kern/assert.h>
+
+struct MountPoint
+{
+    mount_t  mountPoint;
+    uint32_t authCacheDisableCount;
+    int      savedAuthCacheTTL;
+};
+
+static constexpr size_t MaxUsedMountPoints = 4;
+
+static Mutex s_mountPointsMutex;
+static MountPoint s_usedMountPoints[MaxUsedMountPoints];
+
+bool MountPoint_Init()
+{
+    s_mountPointsMutex = Mutex_Alloc();
+    if (!Mutex_IsValid(s_mountPointsMutex))
+    {
+        return false;
+    }
+    
+    return true;
+}
+
+void MountPoint_Cleanup()
+{
+    if (Mutex_IsValid(s_mountPointsMutex))
+    {
+        Mutex_FreeMemory(&s_mountPointsMutex);
+    }
+    
+    for (size_t i = 0; i < MaxUsedMountPoints; ++i)
+    {
+        // by the time this function is called, any provider user clients
+        // should have been terminated & deregistered
+        assert(s_usedMountPoints[i].mountPoint == nullptr);
+        assert(s_usedMountPoints[i].authCacheDisableCount == 0);
+    }
+}
+
+static ssize_t FindMountPoint(mount_t mountPoint)
+{
+    for (size_t i = 0; i < MaxUsedMountPoints; ++i)
+    {
+        if (s_usedMountPoints[i].mountPoint == mountPoint)
+        {
+            return i;
+        }
+    }
+    
+    return -1;
+}
+
+void MountPoint_DisableAuthCache(mount_t mountPoint)
+{
+    Mutex_Acquire(s_mountPointsMutex);
+    {
+        ssize_t mountPointIndex = FindMountPoint(mountPoint);
+        if (mountPointIndex < 0)
+        {
+            mountPointIndex = FindMountPoint(nullptr);
```

Comment:
1. Check if there's already a slot for this mount point.
2. If not, find an empty slot.
3. If that succeeds, fill the slot and use it, otherwise follow the full array logic.

Reply:
The `nullptr` mount point was the non-intuitive part to me at my first pass through this. Maybe a comment here would make this more intuitive?